src/components/va-text-input/va-text-input.tsx

```diff
@@ -120,6 +130,15 @@ export class VaTextInput {
     const describedBy =
       `${this.ariaDescribedby} ${this.error ? 'error-message' : ''}`.trim() ||
       null; // Null so we don't add the attribute if we have an empty string
+    const inputMode =
+      this.inputmode ?
+      this.inputmode :
+      null; // Null so we don't add the attribute if we have an empty string
+    const type =
+      ['email', 'number', 'search', 'tel', 'text', 'url'].includes(this.type) ?
```

Comment:
Can you pull this out into a named variable for clarity? Maybe `allowedInputTypes` or `textInputTypes` or something.